### PR TITLE
Convert string proto to 1d u8

### PR DESCRIPTION
**Pull request by dkalinowski**
- fill remaining 2d padding with zeros

JIRA: CVS-105403

### CODE REVIEW

File: src/predict_request_validation_utils.cpp [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include "kfs_frontend/kfs_grpc_inference_service.hpp"
 #include "kfs_frontend/kfs_utils.hpp"
 #include "modelconfig.hpp"
+#include "precision.hpp"
```
Comment (dkalinowski) [inline src/predict_request_validation_utils.cpp:35]: to remove

File: src/deserialization.hpp [hunk 2]
```diff
@@ -348,20 +355,27 @@ Status deserializePredictRequest(
...
-                    SPDLOG_DEBUG("String input format conversion failed.");
-                    return status;
-                }
-            } else if (isNativeFileFormatUsed(requestInput)) {
-                SPDLOG_DEBUG("Request contains input in native file format: {}", name);
-                status = convertNativeFileFormatRequestTensorToOVTensor(requestInput, tensor, tensorInfo, nullptr);
-                if (!status.ok()) {
-                    SPDLOG_DEBUG("Input native file format conversion failed.");
-                    return status;
+            if (hasString(requestInput)) {
```
Comment (michalkulakowski) [inline src/deserialization.hpp:358]: change to requireProcessing

File: src/deserialization.hpp [hunk 3]
```diff
@@ -415,18 +429,24 @@ Status deserializePredictRequest(
...
+                case TensorInfo::ProcessingHint::STRING_2D_U8:
+                    SPDLOG_DEBUG("Request contains input in 2D string format: {}", name);
+                    RETURN_IF_ERR(convertStringRequestTensorToOVTensor(*requestInputItr, tensor, bufferLocation));
+                    break;
+                case TensorInfo::ProcessingHint::IMAGE:
+                    SPDLOG_DEBUG("Request contains input in native file format: {}", name);
+                    RETURN_IF_ERR(convertNativeFileFormatRequestTensorToOVTensor(*requestInputItr, tensor, tensorInfo, bufferLocation));
+                    break;
+                default:
+                    SPDLOG_DEBUG("String input: {} has no conversion hint to dim_size: {}; precision: {}",
```
Comment (michalkulakowski) [inline src/deserialization.hpp:447]: improve this log please

File: src/tensor_conversion.cpp [hunk 4]
```diff
@@ -462,49 +462,69 @@ static Status convertNativeFileFormatRequestTensorToOVTensor(const TensorType& s
...
+    int64_t totalStringsLength = 0;
+    for (int i = 0; i < batchSize; i++) {
+        totalStringsLength += getBinaryInput(src, i).size();
+    }
+    int64_t width = totalStringsLength + sizeof(uint32_t) * (batchSize + 2);
+    tensor = ov::Tensor(ov::element::Type_t::u8, ov::Shape{static_cast<size_t>(width)});
+    uint32_t* data = reinterpret_cast<uint32_t*>(tensor.data<uint8_t>());
+    data[0] = static_cast<uint32_t>(batchSize);
+    data[1] = 0;  // first string start offset
+    unsigned char* condensedContentStart = tensor.data<unsigned char>() + sizeof(uint32_t) * (batchSize + 2);
```
Comment (michalkulakowski) [inline src/tensor_conversion.cpp:506]: Maybe we can find meaningful name for batchSize + 2 value. Also comment with explanation what is going on in this function would be helpful.

Comment (michalkulakowski) [inline src/tensor_conversion.cpp:506]: condensedStringsStart?

Comment (dkalinowski) [reply src/tensor_conversion.cpp]: done

Comment (dkalinowski) [reply src/tensor_conversion.cpp]: done

File: src/deserialization.hpp (hunk 3 above)
Comment (dkalinowski) [reply src/deserialization.hpp]: done

File: src/deserialization.hpp (hunk 2 above)
Comment (dkalinowski) [reply src/deserialization.hpp]: done

**Pull request merged**